public/app/core/components/SVG/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { v4 as uuidv4 } from 'uuid';
+
+export const getSvgStyle = (svgCode: string) => {
+  const svgStyle = svgCode.match(new RegExp('<style type="text/css">([\\s\\S]*?)<\\/style>'));
+  return svgStyle ? svgStyle[0] : null;
+};
+
+export const getSvgId = (svgCode: string) => {
+  return svgCode.match(new RegExp('<svg.*id\\s*=\\s*([\'"])(.*?)\\1'))?.[2];
```

Comment:
This looks a little scary to maintain 😬 at very least maybe we should consider replacing the `2` with a const explaining what the index represents?